src/main/java/com/avispl/symphony/dal/infrastructure/management/philips/huebridge/PhilipsHueDeviceCommunicator.java

```diff
@@ -0,0 +1,4123 @@
+									oldItems.add(item);
+									roomAndListChildren.put(group, oldItems);
+									nameOfGroupMap.put(bridgeId.get().getId(), roomAndListChildren);
+								}
+							} else {
+								Map<Group, List<Group>> groupAndDevice = nameOfGroupMap.get(bridgeId.get().getId());
+								group.setId(bridgeId.get().getId());
+								group.setType(PhilipsConstant.BRIDGE_HOME);
+								Group item = new Group();
+								item.setId(idLightDevice);
+								item.setType(PhilipsConstant.LIGHT);
+								//Copy add device in room to bridge hone type
+								//Handle case device can't be assigned in room
+								if (groupAndDevice == null) {
+									if (nameOfGroupMap.size() > 0) {
+										for (Entry<String, Map<Group, List<Group>>> maps : nameOfGroupMap.entrySet()) {
+											for (List<Group> groupList : maps.getValue().values()) {
+												for (Group groupItem : groupList) {
+													//copy all device in room tp bridge
+													newItems.add(groupItem);
+												}
+											}
+											//remove all group with room type
+											nameOfGroupMap.remove(maps.getKey());
+										}
+										oldItems = newItems;
+									}
+									oldItems.add(item);
+									roomAndListChildren.put(group, oldItems);
+									nameOfGroupMap.put(bridgeId.get().getId(), roomAndListChildren);
+								} else {
+									oldItems.add(item);
+									roomAndListChildren.put(group, oldItems);
+									groupAndDevice.put(group, oldItems);
+								}
+							}
+						}
+					}
+				}
+				for (Entry<String, Map<Group, List<Group>>> deviceList : nameOfGroupMap.entrySet()) {
+					for (Entry<Group, List<Group>> deviceDetail : deviceList.getValue().entrySet()) {
+						Location location = new Location();
+						location.setGroup(deviceDetail.getKey());
+						location.setItems(deviceDetail.getValue().toArray(new Group[0]));
+						locationList.add(location);
+					}
+				}
+			}
+			if (locationList.isEmpty()) {
+				throw new ResourceNotReachableException("Error creating device automation cannot be empty. Please select the device to create automation");
+			}
+			autoConfiguration.setLocation(locationList.toArray(new Location[0]));
+		}
+
+		//Extract device list by type is room
+		if (TypeOfDeviceEnum.ROOM.getName().equals(type)) {
+			Map<String, String> mapOfDevice = typeAndMapOfDeviceAndValues.get(PhilipsConstant.ROOM);
+			if (mapOfDevice != null) {
+				Map<String, Group> nameOfGroupMap = new HashMap<>();
+				for (Entry<String, String> deviceEntry : mapOfDevice.entrySet()) {
+					Group group = new Group();
+					if (!StringUtils.isNullOrEmpty(deviceEntry.getValue()) && !PhilipsConstant.NONE.equals(deviceEntry.getValue())) {
+						String idDevice = roomNameAndIdMap.get(deviceEntry.getValue());
+						Optional<RoomAndZoneResponse> room = roomList.stream().filter(item -> item.getId().contains(idDevice)).findFirst();
+						if (room.isPresent()) {
+							if (nameOfGroupMap.size() == 0) {
+								group.setId(room.get().getId());
```

Comment:
room.get() could be extracted into a variable to avoid fetching it multiple times

Reply:
I fixed it